test/test_solvers.jl

```diff
@@ -178,8 +179,109 @@ function poisson_ppn_recover_sine_cosine_solution(FT, Nx, Ny, Nz, Lx, Ly, Lz, mx
     isapprox(ϕ, Ψ.(xC, yC, zC); rtol=5e-2)
 end
 
-@testset "Poisson solvers" begin
-    println("Testing Poisson solvers...")
+function can_solve_single_tridiagonal_system(arch, N)
+    ArrayType = array_type(arch)
+
+    a = rand(N-1) |> ArrayType
+    b = 3 .+ rand(N) |> ArrayType  # +3 to ensure diagonal dominance.
+    c = rand(N-1) |> ArrayType
+    f = rand(N) |> ArrayType
+
+    M = Tridiagonal(a, b, c)
+    ϕ_correct = M \ f  # This solve probably invokes scalar CuArray operations on the GPU!
+
+    ϕ = reshape(zeros(N), (1, 1, N)) |> ArrayType
+
+    grid = RegularCartesianGrid(size=(1, 1, N), length=(1, 1, 1))
+    btsolver = BatchedTridiagonalSolver(arch; dl=a, d=b, du=c, f=f, grid=grid)
+
+    solve_batched_tridiagonal_system!(ϕ, arch, btsolver)
+
+    return ϕ[:] ≈ ϕ_correct
```

Comment:
Why do we need the `[:]`?

Reply:
`ϕ_correct` is a 1D array because backslash returns a vector while `ϕ` is a `1×1×Nz` array so I use `ϕ[:]` to flatten it to a 1D vector for `≈` to work.